cpp/include/cudf/strings/string_view.cuh

```diff
@@ -330,35 +330,40 @@ __device__ inline size_type string_view::find(const string_view& str,
   return find(str.data(), str.size_bytes(), pos, count);
 }
 
-__device__ inline size_type string_view::find(const char* str,
-                                              size_type bytes,
-                                              size_type pos,
-                                              size_type count) const
-{
-  const char* sptr = data();
-  if (!str || !bytes) return -1;
-  size_type nchars = length();
+template <bool reverse>
+__device__ inline size_type string_view::find_impl(const char* str,
+                                                   size_type bytes,
+                                                   size_type pos,
+                                                   size_type count) const
+{
+  if (!str || pos < 0) return -1;
+  auto const nchars = length();
   if (count < 0) count = nchars;
-  size_type end = pos + count;
-  if (end < 0 || end > nchars) end = nchars;
-  size_type spos = byte_offset(pos);
-  size_type epos = byte_offset(end);
+  auto const spos = byte_offset(pos);
+  auto const epos = byte_offset(std::min(pos + count, nchars));
```

Comment:
Recently I have adopted a new way to do forward/reverse in my PR:
```
 template <bool forward>
  static __device__ auto element_index_pair_iter(size_type const size)
  {
    if constexpr (forward) {
      return thrust::pair(thrust::make_counting_iterator(0), thrust::make_counting_iterator(size));
    } else {
      return thrust::pair(thrust::make_reverse_iterator(thrust::make_counting_iterator(size)),
                          thrust::make_reverse_iterator(thrust::make_counting_iterator(0)));
    }
  }
```
Then I just use the iterators to scan the range with the same code. Not sure if that can be applied here anyhow.

Reply:
This looks good. There is some technical difficulties including thrust headers in this file where it is used in some jitify scenarios with UDFs. So I'm using raw pointers here for now until whenever that is resolved.